Prepare to add some Dimname/DimnameList overloads

zou3519 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #25585 Name inference for argmax, argmin
* #25582 Name inference for min(Tensor, dim?) / max(Tensor, dim?)
* #25431 Name inference rule for mean, std, var, std_mean, var_mean
* **#25405 Prepare to add some Dimname/DimnameList overloads**

This PR adds schemas to native_functions.yaml, core/Tensor.h, and
core/TensorMethods.h for Dimname/DimnameList overloads for the following
functions:
- min, max, max_values, min_values
- mean, median
- logsumexp, std, var, norm
- std_mean, var_mean

The actual implementations will come in a later PR. I am accumulating
all the addtional schemas and changes to core/{Tensor|TensorMethods}.h
in this PR so that there is only one point of failure for potential
merge conflicts.

Test Plan:
- Check that all pytorch builds still build. [namedtensor ci]

Pull Request resolved: https://github.com/pytorch/pytorch/pull/25405

Differential Revision: [D17116333](https://our.internmc.facebook.com/intern/diff/D17116333)

zou3519 committed:
Prepare to add some Dimname/DimnameList overloads

This PR adds schemas to native_functions.yaml, core/Tensor.h, and
core/TensorMethods.h for Dimname/DimnameList overloads for the following
functions:
- min, max, max_values, min_values
- mean, meadian
- logsumexp, std, var, norm

The actual implementations will come in a later PR. I am accumulating
all the addtional schemas and changes to core/{Tensor|TensorMethods}.h
in this PR so that there is only one point of failure for potential
merge conflicts.

Test Plan:
- Check that all pytorch builds still build. [namedtensor ci]

zou3519 committed:
Update on "Prepare to add some Dimname/DimnameList overloads"


This PR adds schemas to native_functions.yaml, core/Tensor.h, and
core/TensorMethods.h for Dimname/DimnameList overloads for the following
functions:
- min, max, max_values, min_values
- mean, meadian
- logsumexp, std, var, norm

The actual implementations will come in a later PR. I am accumulating
all the addtional schemas and changes to core/{Tensor|TensorMethods}.h
in this PR so that there is only one point of failure for potential
merge conflicts.

Test Plan:
- Check that all pytorch builds still build. [namedtensor ci]

Differential Revision: [D17116333](https://our.internmc.facebook.com/intern/diff/D17116333)

nairbv commented on aten/src/ATen/native/native_functions.yaml:
```diff
@@ -2483,6 +2524,17 @@
+- func: norm.names_ScalarOpt_dim_dtype(Tensor self, Scalar? p, Dimname[1] dim, bool keepdim, *, ScalarType dtype) -> Tensor
+  variants: function, method
+
+- func: norm.names_ScalarOpt_dim(Tensor self, Scalar? p, Dimname[1] dim, bool keepdim=False) -> Tensor
```
I'm a bit unfamiliar with the dot notation being used now (I think it's new?), .. but seems like most suffices are lower case. Does ScalarOpt have to be camelcase here? it looks a bit out of place and not sure if there's any significance to that or if it has to match something else.

zou3519 replied:
It doesn't actually matter what the overload names are. I'm making it so that the overload names match the non-named versions (see norm.ScalarOpt_dim).

zou3519 committed:
Update on "Prepare to add some Dimname/DimnameList overloads"


This PR adds schemas to native_functions.yaml, core/Tensor.h, and
core/TensorMethods.h for Dimname/DimnameList overloads for the following
functions:
- min, max, max_values, min_values
- mean, median
- logsumexp, std, var, norm
- std_mean, var_mean

The actual implementations will come in a later PR. I am accumulating
all the addtional schemas and changes to core/{Tensor|TensorMethods}.h
in this PR so that there is only one point of failure for potential
merge conflicts.

Test Plan:
- Check that all pytorch builds still build. [namedtensor ci]

Pull Request resolved: https://github.com/pytorch/pytorch/pull/25405

Differential Revision: [D17116333](https://our.internmc.facebook.com/intern/diff/D17116333)

zou3519 committed:
Update on "Prepare to add some Dimname/DimnameList overloads"


This PR adds schemas to native_functions.yaml, core/Tensor.h, and
core/TensorMethods.h for Dimname/DimnameList overloads for the following
functions:
- min, max, max_values, min_values
- mean, median
- logsumexp, std, var, norm
- std_mean, var_mean

The actual implementations will come in a later PR. I am accumulating
all the addtional schemas and changes to core/{Tensor|TensorMethods}.h
in this PR so that there is only one point of failure for potential
merge conflicts.

Test Plan:
- Check that all pytorch builds still build. [namedtensor ci]

Pull Request resolved: https://github.com/pytorch/pytorch/pull/25405

Differential Revision: [D17116333](https://our.internmc.facebook.com/intern/diff/D17116333)

zou3519 committed:
Update on "Prepare to add some Dimname/DimnameList overloads"


This PR adds schemas to native_functions.yaml, core/Tensor.h, and
core/TensorMethods.h for Dimname/DimnameList overloads for the following
functions:
- min, max, max_values, min_values
- mean, median
- logsumexp, std, var, norm
- std_mean, var_mean

The actual implementations will come in a later PR. I am accumulating
all the addtional schemas and changes to core/{Tensor|TensorMethods}.h
in this PR so that there is only one point of failure for potential
merge conflicts.

Test Plan:
- Check that all pytorch builds still build. [namedtensor ci]

Pull Request resolved: https://github.com/pytorch/pytorch/pull/25405

Differential Revision: [D17116333](https://our.internmc.facebook.com/intern/diff/D17116333)

zou3519 committed:
Update on "Prepare to add some Dimname/DimnameList overloads"


This PR adds schemas to native_functions.yaml, core/Tensor.h, and
core/TensorMethods.h for Dimname/DimnameList overloads for the following
functions:
- min, max, max_values, min_values
- mean, median
- logsumexp, std, var, norm
- std_mean, var_mean

The actual implementations will come in a later PR. I am accumulating
all the addtional schemas and changes to core/{Tensor|TensorMethods}.h
in this PR so that there is only one point of failure for potential
merge conflicts.

Test Plan:
- Check that all pytorch builds still build. [namedtensor ci]

Pull Request resolved: https://github.com/pytorch/pytorch/pull/25405

Differential Revision: [D17116333](https://our.internmc.facebook.com/intern/diff/D17116333)
